neutron/convert.cpp

```diff
@@ -35,8 +37,22 @@ T convert_generic(T &&d, const Dim from, const Dim to, Op op,
       d.coords().set(from, broadcast(coord, arg.dims()));
     transform_in_place(d.coords()[from], arg, op_);
   }
-  // TODO Should we transform coordinates contained in bucket variables?
-  // 2. Rename dims
+  // 2. Transform coordinates in bucket variables
```

Comment:
I agree with the comment in your description regarding having to change coordinates on the data. At present, there is no link between the two even though conceptually they must be. It would be nice if the conversion only apparently applied to the container coordinate (as in block above).

Maybe there's a way to register the bucket data variable with the relevant coord, so that upon coordinate conversion the item coordinate transformation can also be triggered? 

On the other hand. I'm not sure how major this problem is as I think it's only unit conversion where this is problematic, and that's fairly well encapsulated.

Reply:
> Maybe there's a way to register the bucket data variable with the relevant coord, so that upon coordinate conversion the item coordinate transformation can also be triggered?

This feels like a brittle design which we tend to avoid in scipp.